From 4379168959557663177022f01109b03b47bfcb64 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Mon, 13 Sep 2010 16:27:51 +0800
Subject: [PATCH] ENGR00131397 DMA:fix oops when cat /proc/dma-engine

The origin code does not check the NULL pointer.
The patch adds NULL pointer check.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 arch/arm/plat-mxs/dmaengine.c |   17 ++++++++++++-----
 1 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/arch/arm/plat-mxs/dmaengine.c b/arch/arm/plat-mxs/dmaengine.c
index 0c2485b..2a41644 100644
--- a/arch/arm/plat-mxs/dmaengine.c
+++ b/arch/arm/plat-mxs/dmaengine.c
@@ -604,14 +604,21 @@ static void mxs_dma_proc_seq_stop(struct seq_file *file, void *data)
 
 static int mxs_dma_proc_seq_show(struct seq_file *file, void *data)
 {
-	int result;
-	struct mxs_dma_chan *pchan = (struct mxs_dma_chan *)data;
-	struct mxs_dma_device *pdev = pchan->dma;
-	result = seq_printf(file, "%s-channel%-d	(%s)\n",
+	struct mxs_dma_chan *pchan;
+	struct mxs_dma_device *pdev;
+
+	if (data == NULL)
+		return 0;
+
+	pchan = (struct mxs_dma_chan *)data;
+	pdev = pchan->dma;
+	if (pdev == NULL)
+		return 0;
+
+	return  seq_printf(file, "%s-channel%-d	(%s)\n",
 			    pdev->name,
 			    pchan - mxs_dma_channels,
 			    pchan->name ? pchan->name : "idle");
-	return result;
 }
 
 static const struct seq_operations mxc_dma_proc_seq_ops = {
-- 
1.5.4.4

